### PR TITLE
Promote ExpandedDNSConfig feature to the GA stage

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This promotes the ExpandedDNSConfig feature to the GA stage.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
The `ExpandedDNSConfig` feature has graduated to GA. 'ExpandedDNSConfig' feature was locked to default value and will be removed in v1.30. If you were setting this feature gate explicitly, please remove it now.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2595-expanded-dns-config
```

/uncc all

### CODE REVIEW

**Comment by aojea** [comment]
you first lock the feature gate to GA, and after two releases you can safely remove the code

https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/pkg/features/kube_features.go#L955

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -996,7 +997,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	ExecProbeTimeout: {Default: true, PreRelease: featuregate.GA}, // lock to default and remove after v1.22 based on KEP #1972 update
 
-	ExpandedDNSConfig: {Default: true, PreRelease: featuregate.Beta},
+	ExpandedDNSConfig: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.30
```
[the remark this answers is not in the document]
Comment (gjkim42) [reply pkg/features/kube_features.go:1000]: I think this should remain until +2 release for the version skew policy. However, the code checking the feature gate could be removed in GA.

[the remark this answers is not in the document]
Comment (aojea) [reply pkg/features/kube_features.go:1000]: let's ask @liggitt , he knows all these things, I always get confused 😄

[the remark this answers is not in the document]
Comment (aojea) [reply pkg/features/kube_features.go:1000]: I'm happily stand corrected 😄

**Comment by aojea** [comment]
@gjkim42 remove WIP from the title once you are set

**Comment by gjkim42** [comment]
Thanks, removed it.

**Comment by gjkim42** [comment]
/unassign @SergeyKanzhelev @mrunalp 

Actually, this has more to do with the sig-network.
/assign @aojea

**Comment by aojea** [comment]
/lgtm
/approve

/assign @thockin @derekwaynecarr 
/cc @mrunalp @mikebrow 
FYI for container runtimes implementations

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by gjkim42** [comment]
/hold
as KEP pr has not been merged yet.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by gjkim42** [comment]
/unhold 
as https://github.com/kubernetes/enhancements/pull/3914 has been merged.

**Pull request merged**